Formula/wayland.rb

```diff
@@ -5,9 +5,11 @@ class Wayland < Formula
   sha256 "6dc64d7fc16837a693a51cfdb2e568db538bfdc9f457d4656285bb9594ef11ac"
   license "MIT"
 
+  # Tags with a 90+ patch are unstable (e.g., the 1.9.90 tag is used for the
+  # 1.10.0 beta release) and this regex should only match the stable versions.
   livecheck do
-    url "https://wayland.freedesktop.org/releases.html"
-    regex(/href=.*?wayland[._-]v?(\d+(?:\.\d+)+)\.t/i)
+    url "https://gitlab.freedesktop.org/api/v4/projects/121/releases"
+    regex(/version (\d+(?:\.\d+)+) for the official release/i)
```

Comment:
```suggestion
    url "https://wayland.freedesktop.org/releases.html"
    regex(/href=.*?wayland[._-]v?(\d+\.\d+(?:\.(?:\d|[1-8]\d+)(?:\.\d+)*)?)\.t/i)
```

Your regex is matching the version from release text that may not be guaranteed to be provided with every release and/or maintain that exact format, so I'm not entirely sold on that approach. In general, we only match versions in loose text like this when there isn't a better alternative (and I try to update older checks like this to something better when possible).

The homepage still seems fine in this case and we merely have to adjust the regex to omit versions with a 90+ patch. I've added a suggestion for this, using a modified version of the usual regex pattern we use for this. Namely, the regex was tweaked to also match versions with just a major/minor, since there's some precedent in the separate wayland-protocols versions (e.g., 1.31) and we don't want to miss a version due to being overly strict.

We can tangle with their GitLab in the future if we really need to but checking the homepage is simpler/easier for now.

Reply:
yeah, I would prefer to use gitlab API down the path (but for now, I think release homepage is a good enough source as well). Also the gitlab project id seems a bit tricky to find out.